.github/workflows/testnet.yml

```diff
@@ -12,24 +15,43 @@ jobs:
     runs-on: ubuntu-latest
     steps:
     - uses: actions/checkout@v2
-    - run: >
-        docker login docker.pkg.github.com
-        -u $GITHUB_ACTOR
-        -p ${{ secrets.GITHUB_TOKEN }}
+    - run: docker login docker.pkg.github.com -u $GITHUB_ACTOR -p ${{ secrets.GITHUB_TOKEN }}
     # use previously built image for cache if possible
     - run: docker pull $REPO:latest || true
-    - run: >
-        docker build .
-        -t $REPO:latest
-        -t $REPO:${GITHUB_SHA}
-        --cache-from $REPO:latest
-    - run: docker push $REPO:latest
+    - run: docker pull $REPO:rolling || true
+    - run: docker build . -t $REPO:${GITHUB_SHA} -t $REPO:rolling --cache-from $REPO:rolling --cache-from $REPO:latest
     - run: docker push $REPO:${GITHUB_SHA}
+    - run: docker push $REPO:rolling
 
-  deploy:
+  test-server:
     runs-on: ubuntu-latest
     needs: build
     steps:
+    - uses: actions/checkout@v2
+    - run: docker login docker.pkg.github.com -u $GITHUB_ACTOR -p ${{ secrets.GITHUB_TOKEN }}
+    - run: |
+        cp .env.sample .github/test-server/.env
+        cd .github/test-server
+        export SERVER_IMAGE=$REPO:${GITHUB_SHA}
+        docker-compose up -d
+        sleep 10
+        docker-compose exec -T court_server npm run test:server
```

Comment:
Isn't there a way to have a shared testing workflow that both deployments could depend on instead of having it repeated for both?

Reply:
Yeah that's a good idea, I will add a shared shell script.